metaflow/task.py

```diff
@@ -665,18 +695,24 @@ def run_step(
             if self.ubf_context == UBF_CONTROL:
                 self._finalize_control_task()
 
-            end = time.time() - start
+            # Emit metrics to logger/monitor sidecar implementations
+            duration = time.time() - start
+            with _system_monitor.count("metaflow.task.end"):
+                _system_logger.log(
+                    {
+                        "event_name": "metaflow.task.end",
+                        "event_value": 1,
+                    }
+                )
+                pass
 
-            msg = {
-                "task_id": task_id,
-                "msg": "task ending",
-                "step_name": step_name,
-                "run_id": run_id,
-                "flow_name": self.flow.name,
-                "ts": round(time.time()),
-                "runtime": round(end),
-            }
-            logger.log(msg)
+            _system_monitor.gauge("metaflow.task.duration", duration)
```

Comment:
A measure would be more appropriate here than a gauge.

Reply:
Question is should we be measuring the task duration only until the task "ends" or consider the portion dealing with metadata register as part of the task duration as well? 

If it's the former then it would be difficult to implement using the existing context manager construct - hence why we use the gauge metric.

Reply:
That's a good point so we may need to add non-context specific ones for this. The issue is that a gauge isn't the same as a measure. Gauge are supposed to measure levels of things (like # of machines in a pool, etc) not disparate point measurements like this.